### PR TITLE
Shell Recorder: Improve Portability of Code

### PR DESCRIPTION
**Pull request by sanssecours**
This change adds support for OS X to Shell Recorder. The most problematic part in the attached code is probably `replace_newline_null`:

```sh
tr '\n' '⏎' | tr '\000' '\n'
```

The other changes – removal of `grep -P` – should not be problematic, unless I made a mistake when translating the regex.

### CODE REVIEW

**Comment by KurtMi** [comment]
somehow the travis build job is missing, I am curious if it fixes the mac support :)

**Comment by tom-wa** [comment]
the sed part should work, guess the only reason why i've used the lookbehind regex was to omit the "Warning Number: " and "Error " Strings.

File: tests/shell/shell_recorder/shell_recorder.sh [hunk 1]
```diff
@@ -28,7 +28,8 @@ else
         KDBCOMMAND="kdb"
 fi
 
-
+shopt -s expand_aliases
```
Comment (markus2330) [inline tests/shell/shell_recorder/shell_recorder.sh]: Avoid, use function instead.

Comment (sanssecours) [inline tests/shell/shell_recorder/shell_recorder.sh]: Thanks for the information. I converted the alias to a function it in commit ab772e8.

**Comment by markus2330** [comment]
Thank you!

**Pull request merged**